picard/ui/metadatabox/__init__.py

```diff
@@ -316,76 +292,95 @@ def get_value_as_string(self, item):
                 value = ['']
         return value
 
-    def get_selected_tag_data(self, items):
+    def get_selected_tags(self, items):
         result = TagDiff()
         for item in items:
-            tag = self.tag_diff.tag_names[item.row()]
-            value = ""
-            if item.column() == self.COLUMN_ORIG:
-                value = self.tag_diff.old[tag]
-            elif item.column() == self.COLUMN_NEW:
-                value = self.tag_diff.new[tag]
-            if value:
-                if tag not in result.tag_names:
-                    result.tag_names.append(tag)
-                result.add(
-                    tag=tag,
-                    new=value,
-                    removable=self.tag_diff.status[tag] != TagStatus.NOTREMOVABLE,
-                    readonly=self.tag_diff.status[tag] == TagStatus.READONLY
-                )
+            tag, value = self.get_row_info(item.row())
+            result.add(
+                tag=tag,
+                old=value[self.COLUMN_OLD] if item.column() == self.COLUMN_OLD else None,
+                new=value[self.COLUMN_NEW] if item.column() == self.COLUMN_NEW else None,
```

Comment:
Just a minor thing, but caching the value of `item.column()` wouldn't hurt and make lines a bit shorter:

```python
            column = item.column()
            result.add(
                tag=tag,
                old=value[self.COLUMN_OLD] if column == self.COLUMN_OLD else None,
                new=value[self.COLUMN_NEW] if column == self.COLUMN_NEW else None,
```